app/src/test/java/protect/card_locker/LoyaltyCardViewActivityTest.java

```diff
@@ -263,11 +263,15 @@ private void selectBarcodeWithResult(final Activity activity, final String barco
         assertNotNull(intent);
         assertEquals(intent.getComponent().getClassName(), ScanActivity.class.getCanonicalName());
 
+        // this confuses the test suit, a activity result with request type 2 gets sent to the activity being tested
```

Comment:
@TheLastProject 
this part of the code garbles the intent causing a request type 2 to be sent back to the activity getting tested, it messes with shadowInstrumentation.getTargetAndRequestCodeForIntent that it does not give the original request type back

it didn't break because the barcode capture handling was a catch all https://github.com/TheLastProject/Catima/blob/6ad99d47aa72156860c4def2a329be8c8051b073/app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java#L1090

currently the new activityRequest api internally generates a request type code when a contract is created so, getting a type 2 means it doesn't land on any handler

Is there a reason that it is there? it does not make much sense given we are mocking ScanActivity itself